main.tf

```diff
@@ -1,22 +1,22 @@
 data "template_file" "actiongrouptemplate" {
-  template = "${file("${path.module}/templates/action-group.json")}"
+  template = file("${path.module}/templates/action-group.json")
 }
 
-resource "azurerm_template_deployment" "action-group" {
-  template_body       = "${data.template_file.actiongrouptemplate.rendered}"
-  name                = "${var.short_name}_${var.env}"
-  resource_group_name = "${var.resourcegroup_name}"
+resource "azurerm_resource_group_template_deployment" "action-group" {
+  template_content       = data.template_file.actiongrouptemplate.rendered
+  name                = var.short_name_var.env
+  resource_group_name = var.resourcegroup_name
   deployment_mode     = "Incremental"
 
-  parameters = {
-    location             = "${var.location}"
-    actionGroupName      = "${var.action_group_name}"
-    shortName            = "${var.short_name}"
-    emailReceiverName    = "${var.email_receiver_name}"
-    emailReceiverAddress = "${var.email_receiver_address}"
+  parameters_content = jsonencode({
+    location             = var.location
```

Comment:
these need value wrapped around them:
https://github.com/hmcts/cnp-module-metric-alert/blob/2f15b14bb716442a7e1710a8c67937e911166bf2/main.tf#L16-L31

Reply:
Done, thank you Tim.